src/api-utils/subscanCachedData.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { redisDel, redisGet, redisSet } from '~src/auth/redis';
+
+const EXPIRY_IN_SECONDS = 1800;
+
+export async function getCache(key: string) {
+	const redisData = await redisGet(key);
+	if (redisData) {
+		const props = JSON.parse(redisData);
+		if (!props.error) {
```

Comment:
we won't need the error check here, because we won't be saving error ridden responses to cache